src/main/java/pl/edu/mimuw/chess/Player.java

```diff
@@ -0,0 +1,124 @@
+package pl.edu.mimuw.chess;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+
+import static pl.edu.mimuw.chess.V2.*;
+import static pl.edu.mimuw.chess.ChessBoard.*;
+
+public class Player {
+	private final ChessColor color;
+	private Set<AbstractPiece> pieces;
+	
+	public Player(ChessColor color) {
+		this.color = color;
+		
+		this.pieces = new HashSet<AbstractPiece>();
+		
+		switch (color) {
+			case WHITE:
+				this.pieces.add(new RookPiece(v(7, 0), color));
```

Comment:
chyba trochę zduplikowany kod tu wyszedł - dzięki takiej abstakcji wektora, wlasnie dosc prosto moglbys to uosglnic do metody, ktora tylko by miala rozny argument na 1./2. skladowa wektora (i byc moze cos jeszcze)

Reply:
btw lepiej poslugiwac sie interfejsem, chyba ze potrzebujesz bardzo jakiejs dodatkowej metody z klasy abstrakcyjnej